### PR TITLE
Port RemoveUnusedMembers analyzer/fixer/tests to shared layer (IDE0051/IDE0052)

### PR DESCRIPTION
**Pull request by mavasani**
Recommended to review with xlf file filter: **[Review Link](https://github.com/dotnet/roslyn/pull/41995/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csproj&file-filters%5B%5D=.projitems&file-filters%5B%5D=.resx&file-filters%5B%5D=.vb&file-filters%5B%5D=.vbproj)**

First commit https://github.com/dotnet/roslyn/pull/41995/commits/d2c22964f30ccb77eaae17059c602cd01861487d is just cherry-pick from #41993.
Second commit https://github.com/dotnet/roslyn/pull/41995/commits/57d2e5a6e94e2e3c844e5c93e0ca9742b46c4225 is for the port.

### CODE REVIEW

File: src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs [hunk 1]
```diff
@@ -123,10 +123,10 @@ protected static void AdjustAndAddAppropriateDeclaratorsToRemove(SyntaxNode pare
             }
         }
 
-        private class MyCodeAction : CodeAction.DocumentChangeAction
+        private class MyCodeAction : CustomCodeActions.DocumentChangeAction
```
Comment (CyrusNajmabadi) [inline src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs]: Does this mean telemetry will continue to work?  Or will this explicitly break it?

File: src/EditorFeatures/VisualBasicTest/Utils.vb [hunk 2]
```diff
@@ -112,8 +112,10 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests
             Private ReadOnly _node As T
 
             Sub New(syntaxTree As SyntaxTree, node As T)
+#If Not CODE_STYLE Then
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/Utils.vb]: Why?  Can we not also make these simple helpers available?

File: src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs]: Telemetry should have fully qualified type name of the code action - so as long as we are using special nested types for code actions, we should generate telemetry. I will still look around in code and confirm.

File: src/EditorFeatures/VisualBasicTest/Utils.vb (hunk 2 above)
Comment (mavasani) [reply src/EditorFeatures/VisualBasicTest/Utils.vb]: There is type name conflict in Code style test projects due to Contract being available from both CodeStyle analyzer project and Workspaces project. We can potentially work around this with some IVT massaging, but given the IDE test framework in CodeStyle layer is a temporary workaround, I'd rather just leave this for it to be reverted later.

File: src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Analyzers/Core/CodeFixes/RemoveUnusedMembers/AbstractRemoveUnusedMembersCodeFixProvider.cs]: Yes, we should still get telemetry: https://github.com/dotnet/roslyn/blob/ff9cc22cddbe04895fdcc1be5f4b1552da0648ea/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs#L67-L71

**Pull request merged**